src/components/routing/routingRule.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from 'react'
+import { Modal, Form, Input, Select, Checkbox } from 'antd';
+import FormItemLabel from "../form-item-label/FormItemLabel"
+
+const RoutingRule = (props) => {
+    const { getFieldDecorator } = props.form;
+
+    const [rewriteCheck, setRewriteCheck] = useState(false);
+    const [hostCheck, setHostCheck] = useState(false);
+    const [routeType, setRouteType] = useState("Prefix_Match")
```

Comment:
Why are you keeping hostCheck, routeType, rewriteCheck etc in state. We can even keep them in ant form as well.